### ISSUE TITLE
Release Agda 2.6.3

### ISSUE THREAD

**Issue by jespercockx** [agda/agda#6055]
It would be good to work towards a release of 2.6.3 before the next Agda meeting, i.e. by the end of October.

To get us started, I went through all the issues with milestone 2.6.3 and moved many of them to either the icebox or the (newly created) "later" milestone, which is intended for issues that we are planning to fix in an upcoming release but not the next one. If you can, please take a look at the issues with that new milestone and comment or reassign the milestone if you disagree.

This currently leaves us with 39 issues that still have the 2.6.3 milestone, perhaps it would be good to restart the weekly online developer meetings to discuss who can work on which of these issues and which ones could be postponed. @UlfNorell @andreasabel @nad @plt-amy @Saizan since you are currently assigned to one or more of the issues with milestone 2.6.3, would you be available for a meeting in the coming week, perhaps Wednesday afternoon? (What time zone are you in @plt-amy ?)

**Comment by plt-amy** [comment]
I'm in GMT-3 (America/Sao_Paulo), and Wednesday works for me 🙂

**Comment by jespercockx** [comment]
Let's have a meeting on Wednesday at 15:00 (GMT+2) / 10:00 (GMT-3) then. I'll send you the details in private.

**Comment by jespercockx** [comment]
Here is a list of the current "orphan" issues that do not yet have anyone assigned:
- [x] #4216
- [x] #4985
- [x] #5257
- [x] #5378
- [x] #5525
- [x] #5702
- [x] #5719
- [x] #5754
- [x] #5794
- [x] #5816
- [x] #5900
- [x] #6005
- [x] #6047
- [x] #6059
- [x] #6061

**Comment by jespercockx** [comment]
Next meeting: October 5th

**Comment by Terramorpha** [comment]
Hi, has Agda 2.6.3 been released? I see every issue in the list has been closed, but I can't see Agda 2.6.3 on hackage.

**Comment by jespercockx** [comment]
There are just three open issues left (#5702 #6047 and #6220) so we could start to prepare a release candidate. @asr do you perhaps have time this week to assist with the release?

**Comment by andreasabel** [comment]
There is lots of closed issues without milestone which should potentially be added to 2.6.3, so that they appear in the changelog.  I get started on this.

**Comment by andreasabel** [comment]
> There is lots of closed issues without milestone which should potentially be added to 2.6.3, so that they appear in the changelog. I get started on this.

I am done going through the relevant issues without a milestone and assigned these.  Fingers crossed, we should now not have any issue closed by 2.6.3 which is not on the milestone.

**Comment by andreasabel** [comment]
> There are just three open issues left (#5702 #6047 and #6220) so we could start to prepare a release candidate.

Now there is only #6220 left from this list.  We could start a release candidate as soon as @nad 's PR #6221 has landed in `master`, and the other hot PRs have cooled.  Once we forked a release branch, adding stuff is extra work, so I'd do the fork not too early.

**Comment by andreasabel** [comment]
This is the list of closed issues currently computed by `closed-issues-for-milestone 2.6.3`.  We should check for correct issue descriptions, whether these issues should appear in the changelog, and whether the changelog contains more detailed information when necessary:
- ~~[#394](https://github.com/agda/agda/issues/394): Syntax binders with multi-argument lambdas~~
  _(has changelog)_
- ~~[#957](https://github.com/agda/agda/issues/957): Auto missing support for Copatterns~~
   _(has changelog)_
- ~~[#1625](https://github.com/agda/agda/issues/1625): Remove "--experimental" from "--experimental-lossy-unification"~~
   _(has changelog)_
- ~~[#3714](https://github.com/agda/agda/issues/3714): Rename inc/ouc everywhere~~
   (changelog: [091fdf5163](https://github.com/agda/agda/commit/091fdf51633cbc08fed2cded3069fac02aaf5138)) @plt-amy
- [x] [#4103](https://github.com/agda/agda/issues/4103): Rewrite rule rejected because of projection likeness
- ~~[#4384](https://github.com/agda/agda/issues/4384): Rewrite rules with interval arguments are not supported~~
   _(has changelog)_ 
- [x] [#4506](https://github.com/agda/agda/issues/4506): Lack of unicode support in locale may result in uncaught IOException
   ~~__needs changelog__: new option `--transliterate` @nad~~ 
- ~~[#4701](https://github.com/agda/agda/issues/4701): A compiler for Cubical Agda programs where all cubical parts have been marked as erased~~
   _(has changelog)_
- [x] [#4725](https://github.com/agda/agda/issues/4725): Program rejected due to moved dot pattern
- [x] [#4755](https://github.com/agda/agda/issues/4755): Rewrite rule on constructor uses wrong type for matching
- [x] [#4763](https://github.com/agda/agda/issues/4763): Unquote anonymous copattern involving path
- ~~[#4778](https://github.com/agda/agda/issues/4778): Expansion of ellipsis no longer faithful to what the user wrote~~
- [x] [#4786](https://github.com/agda/agda/issues/4786): Should the parameters be marked as erased in the types of constructors and projections?
   _(has changelog)_
- [x] [#5191](https://github.com/agda/agda/issues/5191): Unifier can use erased variables in non-erased data parameters
- ~~[#5293](https://github.com/agda/agda/issues/5293): C-c C-s (solve constraints) infers the incorrect module to project from~~
- [x] [#5378](https://github.com/agda/agda/issues/5378): Internal error with tactic on record field
- [x] [#5427](https://github.com/agda/agda/issues/5427): Remove support for subtyping for erasure and irrelevance
   _(has changelog)_
- [x] [#5448](https://github.com/agda/agda/issues/5448): Should the predicate be erasable in the subst rule (without-K)
- [x] [#5462](https://github.com/agda/agda/issues/5462): Internal error caused by a REWRITE
- [x] [#5468](https://github.com/agda/agda/issues/5468): Disallow certain forms of pattern matching when an index is erased
- [x] [#5525](https://github.com/agda/agda/issues/5525): Duplicate entries in `executables` file lead to undefined behavior
- [x] [#5551](https://github.com/agda/agda/issues/5551): Panic when showing module contents with pattern synonym
- [x] [#5563](https://github.com/agda/agda/issues/5563): Allow erased names in the type signatures of let-bound definitions
- [x] [#5581](https://github.com/agda/agda/issues/5581): Lexical error with tab character in literate Agda text
- [x] [#5589](https://github.com/agda/agda/issues/5589): Internal error with REWRITE of function from path
- [x] [#5634](https://github.com/agda/agda/issues/5634): Restricting the DOT backend to specific libraries
   _(has changelog)_
- [x] [#5644](https://github.com/agda/agda/issues/5644): One can override .agda-lib files in "sneaky" ways
- [x] [#5661](https://github.com/agda/agda/issues/5661): Cubical does not ensure EQUIVFUN has left inverse
- [x] [#5681](https://github.com/agda/agda/issues/5681): "Panic: uncaught pattern violation" on record with hole sort
- [x] [#5702](https://github.com/agda/agda/issues/5702): Can't case split an `HitInt` with some already existing cases
- [x] [#5715](https://github.com/agda/agda/issues/5715): Proposal: Use `Telescope` for `getContext`, `inContext`, and `extendContext`
- [x] [#5727](https://github.com/agda/agda/issues/5727): Reducing universe levels before checking is not sufficient
- [x] [#5728](https://github.com/agda/agda/issues/5728): Internal error when pattern matching on `...` in with statement without providing a pattern match
- [x] [#5731](https://github.com/agda/agda/issues/5731): Instantiated meta-variables in interfaces?
   _(has changelog)_
- [x] [#5734](https://github.com/agda/agda/issues/5734): Relevance check in reflection
- [x] [#5751](https://github.com/agda/agda/issues/5751): json interaction produces Haskell output for SolveAll
- [x] [#5760](https://github.com/agda/agda/issues/5760): Some code related to Cubical Agda runs also when the K rule is on
- [x] [#5763](https://github.com/agda/agda/issues/5763): Internal parser error using syntax rules
- [x] [#5765](https://github.com/agda/agda/issues/5765): Erasure check failure when pattern matching on refl in erased definition
- [x] [#5769](https://github.com/agda/agda/issues/5769): Parameter arguments of projections of instance-opened record type are not erased
- [x] [#5770](https://github.com/agda/agda/issues/5770): Option to erase record parameters in record module without erasing them as arguments to the type
  _(has changelog)_
- [x] [#5775](https://github.com/agda/agda/issues/5775): JSON interaction produces fully qualified terms
- [x] [#5781](https://github.com/agda/agda/issues/5781): No information about modules in output for -vprofile.modules:10
   _(has changelog)_
- [x] [#5794](https://github.com/agda/agda/issues/5794): `Prelude.!!: index too large`
- [x] [#5801](https://github.com/agda/agda/issues/5801): Make it possible to limit how many times the syntactic equality shortcut is allowed to fail
- [x] [#5823](https://github.com/agda/agda/issues/5823): Loop on eta record
- [x] [#5828](https://github.com/agda/agda/issues/5828): Panic when trying to automatically fill a hole with `-r` in the presence of a pattern synonym.
~~- [#5843](https://github.com/agda/agda/issues/5843): Proposal: new flag `--cubical-compatible` replacing `--without-K`~~
- [x] [#5845](https://github.com/agda/agda/issues/5845): Internal error caused by abstracting `variables`
- [x] [#5848](https://github.com/agda/agda/issues/5848): Internal error with `--confluence-check`
- [x] [#5850](https://github.com/agda/agda/issues/5850): Warn about useless hiding in `variable` declaration
- [x] [#5856](https://github.com/agda/agda/issues/5856): Lambda with irrefutable pattern is not rejected when used on Path
- [x] [#5875](https://github.com/agda/agda/issues/5875): Instance Search breaks Termination Highlighting
- ~~[#5876](https://github.com/agda/agda/issues/5876): GHC 9.0.2 missing warning: -Wunicode-bidirectional-format-characters~~
- [x] [#5891](https://github.com/agda/agda/issues/5891): SizeUniv : SizeUniv is inconsistent
- [x] [#5901](https://github.com/agda/agda/issues/5901): Use emacs --batch mode in agda-mode setup
- [x] [#5920](https://github.com/agda/agda/issues/5920): Erased constructors are not treated correctly
- [x] [#5922](https://github.com/agda/agda/issues/5922): Unexpected failure of termination checking for reflection-generated code
- [x] [#5923](https://github.com/agda/agda/issues/5923): Internal error in rewriting
- ~~[#5930](https://github.com/agda/agda/issues/5930): Support GHC 9.2.3~~
- ~~[#5932](https://github.com/agda/agda/issues/5932): The flag `Werror=unrecognised-warning-flags` does not turn unrecognised warnings on fatal errors~~
- [x] [#5944](https://github.com/agda/agda/issues/5944): Internal error in rewriting with --two-level
- [x] [#5953](https://github.com/agda/agda/issues/5953): Recursor of inductive-inductive type does not pass termination check in Cubical Agda
- [x] [#5955](https://github.com/agda/agda/issues/5955): Composition of Glue Type (Seemingly) Causes Infinite Loop
- [x] [#5956](https://github.com/agda/agda/issues/5956): Cubical Agda crashes when interactively normalizing J for equivalences
- [x] [#5966](https://github.com/agda/agda/issues/5966): Performance issues/opportunities involving hash tables
- [x] [#5989](https://github.com/agda/agda/issues/5989): Internal error when using function whose type contains private tactic
- [x] [#6003](https://github.com/agda/agda/issues/6003): de Bruijn index out of scope when rewriting
- [x] [#6006](https://github.com/agda/agda/issues/6006): Internal error rewriting with holes
- [x] [#6015](https://github.com/agda/agda/issues/6015): Pi types with different domain finiteness should not be equal
- ~~[#6019](https://github.com/agda/agda/issues/6019): Support GHC 9.2.4~~
- [x] [#6022](https://github.com/agda/agda/issues/6022): Private bindings in imported modules defeat check for binding of primIdFace/primIdPath
- [x] [#6042](https://github.com/agda/agda/issues/6042): De Bruijn index out of scope when rewriting without-K
- [x] [#6043](https://github.com/agda/agda/issues/6043): de Bruijn error on unexpected implicit argument
- [x] [#6049](https://github.com/agda/agda/issues/6049): Separate flags for "compatible with univalence" and "compatible with Cubical Agda"
- [x] [#6059](https://github.com/agda/agda/issues/6059): Strange code in the termination checker
- [x] [#6066](https://github.com/agda/agda/issues/6066): The meaning of `pattern` without `no-eta-equality` shoud be documented
- [x] [#6067](https://github.com/agda/agda/issues/6067): Another de Bruijn error in rewriting
- [x] [#6073](https://github.com/agda/agda/issues/6073): Constraint solving does not honour singleton types
- [x] [#6074](https://github.com/agda/agda/issues/6074): piSort/funSort of IUniv should be blocked on the codomain
- [x] [#6076](https://github.com/agda/agda/issues/6076): Minibuffer display for `\;` is strange
- [x] [#6080](https://github.com/agda/agda/issues/6080): A space leak due to absName
- [x] [#6082](https://github.com/agda/agda/issues/6082): Elaborate-and-give does not respect --postfix-projections
- [x] [#6093](https://github.com/agda/agda/issues/6093): Different NaN on M1 mac
- [x] [#6095](https://github.com/agda/agda/issues/6095): Duplicated pattern causes error only when defining an unnamed module
- ~~[#6096](https://github.com/agda/agda/issues/6096): Should n be size tel?~~
- ~~[#6108](https://github.com/agda/agda/issues/6108): Higher constructors can trick the productivity checker~~
   _(has changelog)_
- [x] [#6112](https://github.com/agda/agda/issues/6112): Internal error: non-confluent rewriting to singletons
- ~~[#6128](https://github.com/agda/agda/issues/6128): Stack CI times out for Windows~~
- [x] [#6145](https://github.com/agda/agda/issues/6145): Avoid recomputing `SourceToModule` maps
- [x] [#6205](https://github.com/agda/agda/issues/6205): Internal error with `withReconstructed`
- ~~[#6220](https://github.com/agda/agda/issues/6220): Should README modules still be supported?~~

Issues that get removed from the changelog by adding suitable not-in-changelog tags are ~~striked out~~.
After finalizing this list, I'd like to structure it into features, regression fixes (grouped by Agda version that introduced the regression), and other bug fixes.

**Comment by plt-amy** [comment]
@andreasabel What needs to be done re.
>  We should check for correct issue descriptions

this?

**Comment by andreasabel** [comment]
@plt-amy : I meant that if an issue has a non-descriptive title, go _there_ and edit it (don't edit it in the list above).
And then, some issues that add new features to Agda (rather than being mere bug fixes) need a changelog entry explaining the new feature and how to use it (or link to the docs).

**Comment by nad** [comment]
> * [ ]  [#4506](https://github.com/agda/agda/issues/4506): Lack of unicode support in locale may result in uncaught IOException
>   **needs changelog**: new option `--transliterate` @nad

Do you really want to advertise this hack (which is documented in the user manual)? The option is included as a (not very good) workaround for those who encounter error messages like "commitBuffer: invalid argument (invalid character)". If that error message is printed, then Agda will hopefully inform the user about possible workarounds, including the use of `--transliterate`:

https://github.com/agda/agda/blob/1a89181254a1f7a1f83d0f87258589f54c643521/src/full/Agda/Main.hs#L337-L355

**Comment by andreasabel** [comment]
> Do you really want to advertise this hack (which is documented in the user manual)?

Fine with me if you want to keep it under the radar.

**Comment by asr** [comment]
> @asr do you perhaps have time this week to assist with the release?

Yes. I'll prepare the RC.

**Comment by asr** [comment]
In the [CHANGELOG](https://hackage.haskell.org/package/Agda-2.6.2.2/changelog) for 2.6.2.2 (and other versions too) we included a section with highlights. Could someone add some highlights for 2.6.3?

**Comment by asr** [comment]
> Now there is only #6220 left from this list. We could start a release candidate as soon as @nad 's PR #6221 has landed in `master`, and the other hot PRs have cooled. Once we forked a release branch, adding stuff is extra work, so I'd do the fork not too early.

It seems we are ready for creating the release branch. Right?

**Comment by andreasabel** [comment]
> It seems we are ready for creating the release branch. Right?

There are some PRs left with milestone 2.6.3 which I expect to land very soon.  You could wait for these to save you some cherry-picking and keep a simpler commit history.

There are also 
- [x] some CHANGELOG entries missing still which need adding, and 
- [x] #6226 some new contributors to be added to `team.rst`.

**Comment by nad** [comment]
I just moved issue #5577 to the 2.6.3 milestone.

**Comment by andreasabel** [comment]
Updated list of currently closed issues in for the 2.6.3 changelog (doesn't include #5577 and other to-be-closed issues):

Regressions in Agda 2.6.0
-------------------------

- [#5875](https://github.com/agda/agda/issues/5875): Instance Search breaks Termination Highlighting
- [#6095](https://github.com/agda/agda/issues/6095): Ambiguous pattern synonyms broken with anonymous module

Regressions in Agda 2.6.1
-------------------------

Internal errors triggered by rewriting:

- [#5923](https://github.com/agda/agda/issues/5923): Internal error in rewriting
- [#6003](https://github.com/agda/agda/issues/6003): de Bruijn index out of scope when rewriting
- [#6006](https://github.com/agda/agda/issues/6006): Internal error rewriting with holes
- [#6042](https://github.com/agda/agda/issues/6042): De Bruijn index out of scope when rewriting without-K
- [#6043](https://github.com/agda/agda/issues/6043): de Bruijn error on unexpected implicit argument
- [#6067](https://github.com/agda/agda/issues/6067): Another de Bruijn error in rewriting

Regressions in Agda 2.6.2
-------------------------

- [#4725](https://github.com/agda/agda/issues/4725): Cubical Agda: Program rejected by termination checker due to moved dot pattern
- [#5681](https://github.com/agda/agda/issues/5681): Panic on record declaration with unknown sort
- [#5727](https://github.com/agda/agda/issues/5727): Reducing universe levels before checking is not sufficient
- [#5728](https://github.com/agda/agda/issues/5728): Internal error when pattern matching on `...` in with statement without providing a pattern match
- [#5845](https://github.com/agda/agda/issues/5845): Internal error caused by abstracting `variables`
- [#5922](https://github.com/agda/agda/issues/5922): Failure of termination checking for reflection-generated code due to data projections
- [#5953](https://github.com/agda/agda/issues/5953): Recursor of inductive-inductive type does not pass termination check in Cubical Agda
- [#5955](https://github.com/agda/agda/issues/5955): Composition of Glue Type Causes Infinite Loop

Other bug fixes and new features
---------------------------------
- [#394](https://github.com/agda/agda/issues/394): Syntax binders with multi-argument lambdas
- [#957](https://github.com/agda/agda/issues/957): Auto missing support for Copatterns
- [#1625](https://github.com/agda/agda/issues/1625): Remove "--experimental" from "--experimental-lossy-unification"
- [#3714](https://github.com/agda/agda/issues/3714): Rename inc/ouc everywhere
- [#4103](https://github.com/agda/agda/issues/4103): Rewrite rule rejected because of projection likeness
- [#4384](https://github.com/agda/agda/issues/4384): Rewrite rules with interval arguments are not supported
- [#4506](https://github.com/agda/agda/issues/4506): Lack of unicode support in locale may result in uncaught IOException
- [#4701](https://github.com/agda/agda/issues/4701): A compiler for Cubical Agda programs where all cubical parts have been marked as erased
- [#4755](https://github.com/agda/agda/issues/4755): Rewrite rule on constructor uses wrong type for matching
- [#4763](https://github.com/agda/agda/issues/4763): Cubical Agda: Unquote anonymous copattern involving path
- [#4786](https://github.com/agda/agda/issues/4786): Should the parameters be marked as erased in the types of constructors and projections?
- [#5191](https://github.com/agda/agda/issues/5191): Unifier can use erased variables in non-erased data parameters
- [#5378](https://github.com/agda/agda/issues/5378): Internal error with tactic on record field
- [#5427](https://github.com/agda/agda/issues/5427): Remove support for subtyping for erasure and irrelevance
- [#5448](https://github.com/agda/agda/issues/5448): Should the predicate be erasable in the subst rule (without-K)
- [#5462](https://github.com/agda/agda/issues/5462): Internal error caused by a REWRITE on a projection-like function
- [#5468](https://github.com/agda/agda/issues/5468): Disallow certain forms of pattern matching when an index is erased
- [#5525](https://github.com/agda/agda/issues/5525): Duplicate entries in `executables` file lead to undefined behavior
- [#5551](https://github.com/agda/agda/issues/5551): Panic when showing module contents with pattern synonym
- [#5563](https://github.com/agda/agda/issues/5563): Allow erased names in the type signatures of let-bound definitions
- [#5581](https://github.com/agda/agda/issues/5581): Lexical error with tab character in literate Agda text
- [#5589](https://github.com/agda/agda/issues/5589): Internal error with REWRITE of function from path
- [#5634](https://github.com/agda/agda/issues/5634): Restricting the DOT backend to specific libraries
- [#5644](https://github.com/agda/agda/issues/5644): Library (`.agda-lib`) files below the project root
- [#5661](https://github.com/agda/agda/issues/5661): Cubical does not ensure EQUIVFUN has left inverse
- [#5702](https://github.com/agda/agda/issues/5702): Can't case split an `HitInt` with some already existing cases
- [#5715](https://github.com/agda/agda/issues/5715): Reflection: Use `Telescope` for `getContext`, `inContext`, and `extendContext`
- [#5731](https://github.com/agda/agda/issues/5731): Performance: Instantiated meta-variables in interface files
- [#5734](https://github.com/agda/agda/issues/5734): Relevance check in reflection
- [#5751](https://github.com/agda/agda/issues/5751): json interaction produces Haskell output for SolveAll
- [#5760](https://github.com/agda/agda/issues/5760): Some code related to Cubical Agda runs also when the K rule is on
- [#5763](https://github.com/agda/agda/issues/5763): Internal parser error using syntax rules
- [#5765](https://github.com/agda/agda/issues/5765): Erasure check failure when pattern matching on refl in erased definition
- [#5769](https://github.com/agda/agda/issues/5769): Parameter arguments of projections of instance-opened record type are not erased
- [#5770](https://github.com/agda/agda/issues/5770): Option to erase record parameters in record module without erasing them as arguments to the type
- [#5775](https://github.com/agda/agda/issues/5775): JSON interaction produces fully qualified terms
- [#5781](https://github.com/agda/agda/issues/5781): New flag `--profile` replaces verbosity option `-vprofile`
- [#5794](https://github.com/agda/agda/issues/5794): Agsy/Auto crashes with `Prelude.!!: index too large`
- [#5801](https://github.com/agda/agda/issues/5801): Make it possible to limit how many times the syntactic equality shortcut is allowed to fail
- [#5823](https://github.com/agda/agda/issues/5823): Singleton check loops on recursive eta record
- [#5828](https://github.com/agda/agda/issues/5828): Agsy/Auto panics with `-r` in the presence of a pattern synonym
- [#5848](https://github.com/agda/agda/issues/5848): Internal error with `--confluence-check`
- [#5850](https://github.com/agda/agda/issues/5850): Warn about useless hiding in `variable` declaration
- [#5856](https://github.com/agda/agda/issues/5856): Lambda with irrefutable pattern is not rejected when used on Path
- [#5891](https://github.com/agda/agda/issues/5891): SizeUniv : SizeUniv is inconsistent
- [#5901](https://github.com/agda/agda/issues/5901): Use emacs --batch mode in agda-mode setup
- [#5920](https://github.com/agda/agda/issues/5920): Erased constructors skipped in modality check
- [#5944](https://github.com/agda/agda/issues/5944): Internal error in rewriting with --two-level
- [#5956](https://github.com/agda/agda/issues/5956): Cubical Agda crashes when printing empty system
- [#5966](https://github.com/agda/agda/issues/5966): Performance enhancement: switching to `vector-hashtables` library
- [#5989](https://github.com/agda/agda/issues/5989): Dead-code elimination crashes function with private tactic argument
- [#6015](https://github.com/agda/agda/issues/6015): Pi types and Partial types should not be considered inter-convertible
- [#6022](https://github.com/agda/agda/issues/6022): Private bindings in imported modules defeat check for binding of primIdFace/primIdPath
- [#6049](https://github.com/agda/agda/issues/6049): Separate flags for "compatible with univalence" and "compatible with Cubical Agda"
- [#6059](https://github.com/agda/agda/issues/6059): Non-terminating function over tuples passed with `--termination-depth=2`
- [#6066](https://github.com/agda/agda/issues/6066): Document the meaning of `pattern` without `no-eta-equality`
- [#6073](https://github.com/agda/agda/issues/6073): Constraint solving does not honour singleton types
- [#6074](https://github.com/agda/agda/issues/6074): piSort/funSort of IUniv should be blocked on the codomain
- [#6076](https://github.com/agda/agda/issues/6076): Agda input mode (emacs): Minibuffer display for `\;` is strange
- [#6080](https://github.com/agda/agda/issues/6080): A space leak due to absName
- [#6082](https://github.com/agda/agda/issues/6082): Elaborate-and-give does not respect --postfix-projections
- [#6093](https://github.com/agda/agda/issues/6093): Different NaN on M1 mac
- [#6108](https://github.com/agda/agda/issues/6108): Higher constructors can trick the productivity checker
- [#6112](https://github.com/agda/agda/issues/6112): Internal error: non-confluent rewriting to singletons
- [#6145](https://github.com/agda/agda/issues/6145): Preformance enhancement for large projects: Avoid recomputing `SourceToModule` maps
- [#6194](https://github.com/agda/agda/issues/6194): The GHC backend sometimes compiles the wrong modules when --interaction is used
- [#6205](https://github.com/agda/agda/issues/6205): Internal error with `withReconstructed`

**Comment by nad** [comment]
Should issues that are mentioned explicitly in the changelog be included in the list that is added during the release process?

**Comment by jespercockx** [comment]
Usually we apply the label `documented-in-changelog` to them, which implies that they are not listed in the issue list.

**Comment by andreasabel** [comment]
> Usually we apply the label `documented-in-changelog` to them, which implies that they are not listed in the issue list.

Ok, didn't remember this trick.  ~~I can apply this label accordingly to the issues with _(has changelog)_.~~ 
EDIT: seems like someone else is already at it.  That someone else, please also strike them out in https://github.com/agda/agda/issues/6055#issuecomment-1289436327.

**Comment by nad** [comment]
> Usually we apply the label `documented-in-changelog` to them, which implies that they are not listed in the issue list.

I thought that enhancements were not part of the list, because enhancements are typically documented in the changelog.

Anyway, I've added `documented-in-changelog` too all issues that were documented in the changelog.

> That someone else, please also strike them out in [#6055 (comment)](https://github.com/agda/agda/issues/6055#issuecomment-1289436327).

Isn't that list automatically generated? I don't see the point of that exercise.

**Comment by andreasabel** [comment]
> Anyway, I've added `documented-in-changelog` too all issues that were documented in the changelog.

Thanks.

> > That someone else, please also strike them out in [#6055 (comment)](https://github.com/agda/agda/issues/6055#issuecomment-1289436327).
> 
> Isn't that list automatically generated? I don't see the point of that exercise.

That list was originally automatically generated, but now is a manually maintained worklist.
The point of the exercise is to show in one glance that nothing was forgotten.  E.g. you forgot to apply `documented-in-changelog` to #3714.

**Comment by nad** [comment]
Issue #3714 is not mentioned in the changelog.

**Comment by andreasabel** [comment]
Here: https://github.com/agda/agda/blob/6b5511167d92d70fd73909b4dfe7a454f7eb09fc/CHANGELOG.md?plain=1#L216-L218

**Comment by andreasabel** [comment]
For my part, I consider the _manual_ CHANGELOG editing done for now.  (The generated list of closed issues has to be added still once it is complete.)

**Comment by andreasabel** [comment]
Our hopefully final developer meeting before the 2.6.3 RC is tomorrow Thu 27 Oct at 1.30pm Central European Summer Time at Chalmers in EDIT 6128 and online at

  https://gu-se.zoom.us/j/66481990570

@plt-amy @nad @UlfNorell @gallais @asr @jespercockx @L-TChen etc...

**Comment by gallais** [comment]
Tomorrow is St Andrews' open day so I will unfortunately not be able to join.
Nothing is assigned to me AFAIK so hopefully my absence won't be a problem! 😃

**Comment by jespercockx** [comment]
I've added a highlights section to the top of the Changelog.

**Comment by andreasabel** [comment]
> I've added a highlights section to the top of the Changelog.

Should it say something about @Saizan 's work on supporting indexed families in Cubical Agda?
@plt-amy : Maybe you could a sentence pointing to that?

**Comment by jespercockx** [comment]
Which version of the standard library will be the recommended one for Agda 2.6.3? Does the current `master` branch work or should there be a new release of the standard library too to ensure compatibility? If this question is resolved, we could perhaps add a note to the changelog to indicate which version of the library can be used. (Also, summoning @MatthewDaggitt to keep him updated on this.)

**Comment by nad** [comment]
I don't think the current master branch works, there was some change related to `primFloatToWord64`.

I also noticed that the standard library doesn't export the `pattErr` constructor from `Agda.Builtin.Reflection`, that should presumably be fixed.

**Comment by MatthewDaggitt** [comment]
So this is a problem. The `master` branch has a lot of changes related to v2.0 that we're not quite ready to release. The `experimental` branch is supposed to be clean, but last month in https://github.com/agda/agda-stdlib/pull/1843, @UlfNorell commented that it was "out-of-date" and in a moment of complete madness, I merged `master` into it :sob: 

So we don't have a clean experimental from which to make a compatible release with. I think I'm going to have to go back to the `v1.7.1` and cherry-pick only the things that should be on `experimental`....

**Comment by MatthewDaggitt** [comment]
And then make a release for `v1.7.2` to go with it....

**Comment by MatthewDaggitt** [comment]
As discussed I've created a `release-v1.7.2` for the standard library and opened a PR switching to it (https://github.com/agda/agda/pull/6260). May require a couple of CI iterations to check that I've managed to cherry-pick everything of interest.

**Comment by L-TChen** [comment]
Should we mark breaking changes explicitly in the changelog? I'd like to have a warning, say [**Breaking**], added to each item so, for example, there would be a short paragraph about `getContext` and `inContext` in the changelog like

> * [**Breaking**] The reflection primitives `getContext` and `inContext` use a nominal context
  `List (Σ String λ _ → Arg Type)` instead of  `List (Arg Type)` for printing
  type information better. Similarly, `extendContext` takes an extra argument
  of type `String`.

It is important to make the warning easy to see and find, so making it structured might be better.

**Comment by andreasabel** [comment]
@L-TChen : Yes this is a good idea!  Please go ahead!

**Comment by andreasabel** [comment]
@asr: Just on time, as of 1 Nov we have no open PR for 2.6.3.  I think it is an opportunity to make a release candidate now!

**Comment by asr** [comment]
I'm starting right now.

**Comment by asr** [comment]
I just created the `release-2.6.3` branch based on 42a5683c65c2fad64a7feabc1ec3cdf3b7fe9b2b.

**Comment by L-TChen** [comment]
> @L-TChen : Yes this is a good idea! Please go ahead!

I created a PR #6275 for this. Hope it can be checked and merged before the official release.

**Comment by asr** [comment]
Why are not CI running on the `release-2.6.3` branch?

**Comment by asr** [comment]
It seems CI didn't run on the `release-2.6.3` branch because there weren't changes respect to `master`. CI worked after cherry-picking a commit into the branch.

**Comment by andreasabel** [comment]
We are in a kind of limbo state atm:
1. `release-2.6.3` already exists for the release
2. `master` is not yet ready for commits aimed at 2.6.4, because e.g. the 2.6.3 CHANGELOG still sits on `master`

Because of 2., `master` cannot be used for 2.6.4 development and only contains 2.6.3 stuff. So the `release-2.6.3` branch isn't needed yet.
I think when `release-2.6.3` is definitely forked off, this step should also make the `master` ready for `2.6.4` development, e.g., the CHANGELOG should be moved to `doc/release-notes/2.6.3.md`, the Agda version be bumped etc.  
There is no terrible rush for that, this can happen on Monday or Tuesday, but it should happen a bit before the Agda meeting starts.
@asr : What is the time plan for the RC?

**Comment by asr** [comment]
Please note that we first will release a RC and after we will release the final version, so at the moment `master` is only for 2.6.3.

> I think when `release-2.6.3` is definitely forked off, this step should also make the `master` ready for `2.6.4` development, e.g., the CHANGELOG should be moved to `doc/release-notes/2.6.3.md`, the Agda version be bumped etc.

Right. We always make that after the final release (no between the RC and the final version).

> @asr : What is the time plan for the RC?

I'm planning make the release of the RC this weekend.

> So the `release-2.6.3` branch isn't needed yet.

Mmm, Ok. I'll delete the branch and I'll create it again this weekend.

**Comment by asr** [comment]
The `release-2.6.3` branch is frozen.

**Comment by asr** [comment]
I'll release the RC in a couple of hours. We are planning release the final version *after* the Agda Meeting. Right?

**Comment by asr** [comment]
The RC was [upload](https://hackage.haskell.org/package/Agda-2.6.2.2.20221106/candidate) to Hackage.

I'll make the announce in a couple of minutes.

The `release-2.6.3` branch is unfrozen. That is, you can cherry-pick or fast-word your new commits for 2.6.3 in `master` into the `release-2.6.3` branch.

**Comment by asr** [comment]
Argh!, I just noted #6300. I'll fix it and upload a new RC.

**Comment by asr** [comment]
The fixed RC was [upload](https://hackage.haskell.org/package/Agda-2.6.2.2.20221106/candidate) to Hackage.

**Comment by andreasabel** [comment]
Great, thanks for the RC!

> > I think when `release-2.6.3` is definitely forked off, this step should also make the `master` ready for `2.6.4` development, e.g., the CHANGELOG should be moved to `doc/release-notes/2.6.3.md`, the Agda version be bumped etc.
> 
> Right. We always make that after the final release (no between the RC and the final version).

This year, we need `master` to be ready for new stuff during the Agda meeting with starts on Thursday.  Thus, this time we need to depart from our custom.

**Comment by asr** [comment]
> This year, we need `master` to be ready for new stuff during the Agda meeting with starts on Thursday. Thus, this time we need to depart from our custom.

Ok. I bumped the version of `master` to 2.6.4. For reducing the cherry-pick we have two CHANGELOGs: In `master` we have the CHANGELOG for 2.6.4 and in `release-2.6.3` we have the CHANGELOG for 2.6.3.

**Comment by nad** [comment]
> For reducing the cherry-pick we have two CHANGELOGs: In `master` we have the CHANGELOG for 2.6.4 and in `release-2.6.3` we have the CHANGELOG for 2.6.3.

I guess you plan to later commit the completed changelog to the `master` branch in such a way that `git blame` will not give useful information. I would have preferred if you had moved the changelog to `doc/release-notes` in one commit and added the new changelog in a different commit, because I want to be able to use `git blame` on the changelog to see (roughly) which commit introduced a certain feature.

I found [a way](https://stackoverflow.com/a/57005385) to make `git blame` work for the renamed changelog. I'll prepare a pull request.

**Comment by asr** [comment]
> I would have preferred if you had moved the changelog to `doc/release-notes` in one commit and added the new changelog in a different commit, because I want to be able to use `git blame` on the changelog to see (roughly) which commit introduced a certain feature.

For Agda 2.6.3 at the moment, you can use `git blame` on the `CHANGELOG` on the `release-2.6.3` branch. After the final release, you could use `git blame` on `doc/release-notes/2.6.3.md` on `master`.

**Comment by nad** [comment]
> For Agda 2.6.3 at the moment, you can use `git blame` on the `CHANGELOG` on the `release-2.6.3` branch. After the final release, you could use `git blame` on `doc/release-notes/2.6.3.md` on `master`.

When I look at `doc/release-notes/2.6.1.md` it looks as if you wrote most of it.

**Comment by nad** [comment]
Please use `doc/release-notes/2.6.4.md` for changes that should not be part of Agda 2.6.3 (until Agda 2.6.3 has been released), and `CHANGELOG.md` for changes that should be part of Agda 2.6.3.

**Comment by asr** [comment]
> When I look at `doc/release-notes/2.6.1.md` it looks as if you wrote most of it.

It was due to #4705 which was fixed in 2.6.2.

**Comment by nad** [comment]
OK, thanks. However, I still think the current situation is better than the previous one, because now you can (hopefully) fix things that should go into 2.6.3 and edit the changelog in a single commit which can then be cherry-picked, rather than having to prepare a separate commit with the changes to the changelog.

**Comment by asr** [comment]
> OK, thanks. However, I still think the current situation is better than the previous one, because now you can (hopefully) fix things that should go into 2.6.3 and edit the changelog in a single commit which can then be cherry-picked, rather than having to prepare a separate commit with the changes to the changelog.

I agree. Thank you for your solution.

**Comment by asr** [comment]
I realized I didn't include the `agda-mode` program in RC1 (sorry). Are we planning to release a second release candidate?

**Comment by andreasabel** [comment]
> I realized I didn't include the `agda-mode` program in RC1 (sorry). Are we planning to release a second release candidate?

Yes, I propose to release a second RC, once we have completed the milestone again.

**Comment by andreasabel** [comment]
> I realized I didn't include the `agda-mode` program in RC1 (sorry).

Isn't it included when you do `cabal sdist`?

**Comment by asr** [comment]
> Isn't it included when you do `cabal sdist`?

When I removed the `test-suite` from `Agda.cabal` I also removed `agda-mode` from `Agda.cabal`, so it wasn't included by `cabal sdist`.

**Comment by andreasabel** [comment]
Ah, ok, I see, that explains it.  No problem.

**Comment by andreasabel** [comment]
@asr: Now that the relevant PRs for 2.6.3 have landed (there is only a documentation PR open now), could you please release RC2?  

P.S.: We need to make sure that all recently merged PRs for 2.6.3 are also on `release-2.6.3`, see e.g. https://github.com/agda/agda/pull/6313#issuecomment-1324748228.

P.P.S.: The `closed-issues-for-milestone` may have to be run again.

**Comment by asr** [comment]
I'll freeze the branch later.

**Comment by andreasabel** [comment]
I added new commits to `release-2.6.3`, partially because CI broke (our recurring issue #5258).
Just now, there are no open PRs for 2.6.3.

**Comment by nad** [comment]
If all changes have landed, then I can run some benchmarks in the next few days (see #6049).

**Comment by andreasabel** [comment]
@asr: Can you say when you have time to release RC2?

**Comment by asr** [comment]
I'll start to prepare RC2 right now. The `release-2.6.3` branch is frozen.

**Comment by nad** [comment]
I've created a pull request with some changes to the changelog (#6365).

**Comment by asr** [comment]
> I've created a pull request with some changes to the changelog (#6365).

The current status of that PR is 'changes required'. I'll release RC2 in a couple of hours. You will be able to cherry-pick the PR into the release branch after the release of RC2.

**Comment by asr** [comment]
The RC2 was [upload](https://hackage.haskell.org/package/Agda-2.6.2.2.20221128/candidate) to Hackage.

The release-2.6.3 branch is unfrozen. That is, you can cherry-pick your new commits for 2.6.3 in `master` into the `release-2.6.3` branch.

**Comment by jespercockx** [comment]
Possible regression in the RC that might still need to be fixed: https://github.com/agda/agda/issues/6124#issuecomment-1340939129

**Comment by jespercockx** [comment]
I have a fix for #6406 (pushed to @nad's PR at https://github.com/agda/agda/pull/6405), but unfortunately it's somehow causing a heap exhaustion on the cubical library. Specifically this patch seems to be causing the problem: https://github.com/agda/agda/pull/6405/commits/1cbeaa36fae356fc0c3048a0cc02fee614f0fdec 
I have zero idea what's causing the issue in that patch however, so if someone could help me pin it down that would be really appreciated.

**Comment by andreasabel** [comment]
@jespercockx wrote:
> it's somehow causing a heap exhaustion on the cubical library.

Playing with the projection parameters can easily balloon the complexity (this is why we cannot get rid of the projection-likeness feature).  Do you have some benchmarks how your patch affects performance otherwise?

**Comment by jespercockx** [comment]
I ran the std-lib-test before and after my change, and the difference looks quite minimal:
- Total time went from 317,552ms to 329,223ms (+3.6%)
- Total allocation went from 627,740,634,744 bytes to 633,229,825,496 bytes (+0.8%)
- Total memory usage went from 1926 MiB to 1912 MiB (-0.8%)

**Comment by andreasabel** [comment]
>     * Total time went from 317,552ms to 329,223ms (+3.6%)

Maybe not terribly much, but this is paid by everyone, even without using `--erasure` (as this option isn't available yet, cannot switch this off).
Any changes on projection-heavy benchmarks? (`/benchmark/proj/`)

**Comment by jespercockx** [comment]
I've merged the PR, after removing the commit that caused the performance issue. So now we should be all set for the release (or do we need a third RC?)

EDIT: wait no, I missed that https://github.com/agda/agda/issues/6124 is still open too.

EDIT 2: wait no, my edit was confused: the PR did solve the regression part of #6124, so it is no longer blocking the release.

**Comment by jespercockx** [comment]
I'll cherry-pick the fix for the regression part of #6124 on the release branch.

**Comment by jespercockx** [comment]
Cherry-picking done.

**Comment by asr** [comment]
I'll be away from a computer until December 29. I could start the RC3 (or final release) process on December 30.

**Comment by andreasabel** [comment]
@asr Great!  If you do not mind, let's have a (hopefully) final RC.

**Comment by asr** [comment]
I'm starting the process for releasing RC3. The branch `release-2.6.3` is frozen.

**Comment by asr** [comment]
The [RC3](https://hackage.haskell.org/package/Agda-2.6.2.2.20230105/candidate) was released.

The `release-2.6.3` branch is unfrozen. That is, you can cherry-pick your new commits for 2.6.3 in `master` into the `release-2.6.3` branch.

**Comment by andreasabel** [comment]
@asr: Before the next regression shows up, we should release!  Can you make it happen now?

**Comment by asr** [comment]
I'll make the release. The `release-2.6.3` branch is frozen.

**Comment by asr** [comment]
I'm holding the release because #5644, so the `release-2.6.3` branch is unfrozen.

**Comment by andreasabel** [comment]
> I'm holding the release because #5644, so the `release-2.6.3` branch is unfrozen.

@asr: Please proceed with the release.  
This is only a documentation issue, and our documentation is far from perfect, so an inaccuracy in the documentation shouldn't halt the release at this point.

**Comment by asr** [comment]
OK. The `release-2.6.3` branch is frozen again.

**Comment by nad** [comment]
I'm working on the documentation.

**Comment by nad** [comment]
I've prepared a pull request (#6471).

**Comment by asr** [comment]
@nad, I'll wait that you fix the documentation. Let me know when you finish.

**Comment by nad** [comment]
I have cherry-picked the change for the release-2.6.3 branch.

**Comment by asr** [comment]
I'm trying of update the std-lib commit (`make fast-forward-std-lib`) in both `master` and `release-2.6.3`, but I'm getting the following error:

```
$ make benchmark-without-logs
...
Running benchmark monad
make[1]: *** [Makefile:44: logs/20230124-14.22-laptop/monad] Error 42
...

$ cat benchmark/logs/20230124-14.22-laptop/monad
...
Failed to find source of module Effect.Monad in any of the following locations:
...
when scope checking the declaration
  open import Effect.Monad
```

The problem is due to `Effect.Monad` is not in the [release-v1.7.2](https://github.com/agda/agda-stdlib/tree/release-v1.7.2) branch of the standard library. This seems that our CI is not "complete" enough because this problem should have been detected before.

**Comment by asr** [comment]
I filed https://github.com/agda/agda-stdlib/issues/1915.

**Comment by nad** [comment]
In commit b1a6b6872de4604e6133479d84233114205fe209 I changed the standard library commit from https://github.com/agda/agda-stdlib/commit/b85c5d19ceeaaa22c57c138396141b7270218eb1 to https://github.com/agda/agda-stdlib/commit/95270b78d8d6ded67fcdf94095a056a0245de547. Issue #5611 related to some new code in the standard library, and I wanted to ensure that this code was tested. However, I don't think this code is intended to be released along with Agda 2.6.3.

@MatthewDaggitt, is there a branch of the standard library which is intended to be released soon, and which type-checks using the most recent release candidate? This branch should include the last few changes on the experimental branch, and I don't think the [release-v1.7.2](https://github.com/agda/agda-stdlib/tree/release-v1.7.2) branch includes everything.

**Comment by MatthewDaggitt** [comment]
So the `release-v1.7.2` branch is supposed to the branch that releases with 2.6.3. What I hadn't realised is that people have been pushing to `experimental` without going opening a PR first, which means no one on the standard library library side of things was aware of these commits and therefore they haven't been included in `release-v1.7.2`.

**Comment by MatthewDaggitt** [comment]
Does anyone have any idea which commits they think are missing? I'm confused about why the commits on `experimental` and those on `release-v1.7.2` don't align. 

Looks like:
- https://github.com/agda/agda-stdlib/commit/0e1e1e1167d045e534c7d5e845a37caad2b60354 and the two commits after it?

**Comment by andreasabel** [comment]
@nad: Since `v1.7.2` of the standard library is the one intended for `2.6.3`, the solution I see is to set the `std-lib` submodule to this version and revert your changes to `benchmark/monad` on the `release-2.6.3` branch.

@MatthewDaggitt wrote in https://github.com/agda/agda-stdlib/issues/1915#issuecomment-1404636404 : 
> These modules [`Effect.Monad.*`] should not exist in `v1.7.2`.... they're being introduced in `v2.0`

**Comment by nad** [comment]
> Does anyone have any idea which commits they think are missing? I'm confused about why the commits on `experimental` and those on `release-v1.7.2` don't align.

Variants of the following commits should be included (perhaps some of the changes are already included):

* https://github.com/agda/agda-stdlib/commit/e873a786a3cfe76367a5d0a1d9b2da3c0cdf6af1
* https://github.com/agda/agda-stdlib/commit/9df6d6be3921875115d582a78e3bb87b777039d1
* https://github.com/agda/agda-stdlib/commit/0e1e1e1167d045e534c7d5e845a37caad2b60354
* https://github.com/agda/agda-stdlib/commit/c5f42e1fb86b964dfe2558e103f2f4f662e553b3
* https://github.com/agda/agda-stdlib/commit/81a924e41d24669a8935cc1b7168a96f0087ac21

**Comment by nad** [comment]
> @nad: Since `v1.7.2` of the standard library is the one intended for `2.6.3`, the solution I see is to set the `std-lib` submodule to this version and revert your changes to `benchmark/monad` on the `release-2.6.3` branch.

We could do this once the release-v1.7.2 branch has been finalised. However, I don't think it is important that the standard library submodule points to what will become the next version of the standard library. What is important is that it is possible to release a new version of the standard library soon. @MatthewDaggitt, have you tested the standard library release candidate (if there is one) using the most recent Agda release candidate?

**Comment by asr** [comment]
What is weird is that in both `master` and  `release-2.6.3` the submodule commits are the [same](https://github.com/agda/agda/blob/fbc57152f592f726df1426bd351474665c5c6b1d/.gitmodules) and the missing modules had not been detected in `master`. Anyway, since the `Effect` modules don't belong to next release of the standard library (see [comment](https://github.com/agda/agda-stdlib/issues/1915#issuecomment-1404636404)), we should revert the changes in `release-2.6.3` and change to the `experimental` branch of the standard library in `master`. @nad, could you revert the changes in `release-2.6.3`?

**Comment by nad** [comment]
I could, but I don't really see the point of this. I don't think it is important that the std-lib-succeed tests work with version 1.7.2 of the standard library.

I guess that you encountered the problem reported above by changing the standard library submodule so that it points to the release-v1.7.2 branch. You encountered a problem in the std-lib-succeed test, so presumably the std-lib test succeeded. I don't think we need to rerun that test before releasing Agda. I suggest that you use the same commit for the standard library on the release branch as on the master branch, rerun the tests and release Agda.

I think we should consider what to do in the future: which branch of the standard library, if any, should the standard library submodule point to? I want to test new features of Agda, and thus I want to include new code that is added to the standard library fairly soon after it has been added, even if it will not be part of the next (or any) release.

**Comment by MatthewDaggitt** [comment]
Thanks. I'm a bit busy atm, but I will have the time on Sunday and bring branch `v1.7.2` up to date and get it pointing at the `release-2.6.3` branch.

> I think we should consider what to do in the future: which branch of the standard library, if any, should the standard library submodule point to? I want to test new features of Agda, and thus I want to include new code that is added to the standard library fairly soon after it has been added, even if it will not be part of the next (or any) release.

Edited: sorry what I originally typed was complete nonsense.

Is this not the `experimental` branch you're after?

**Comment by MatthewDaggitt** [comment]
> Thanks. I'm a bit busy atm, but I will have the time on Sunday and bring branch v1.7.2 up to date and get it pointing at the release-2.6.3 branch.

I should say that if its urgent, some else is welcome to update it in the meantime.

**Comment by asr** [comment]
> I could, but I don't really see the point of this. I don't think it is important that the std-lib-succeed tests work with version 1.7.2 of the standard library.

My goal isn't test the version 1.7.2 of the standard library with Agda 2.6.3 but the other way around. If that version of the standard library has new commits, we should use them for testing Agda 2.6.3. I'll try to update the standard library commit and fix the problems on the `release-2.6.3` branch, later.

**Comment by asr** [comment]
> Thanks. I'm a bit busy atm, but I will have the time on Sunday and bring branch `v1.7.2` up to date and get it pointing at the `release-2.6.3` branch.

Great! However note that it won't fix the current issue with the `benchmark-without-logs` test.

**Comment by asr** [comment]
> I'll try to update the standard library commit and fix the problems on the `release-2.6.3` branch, later.

I updated the commits of the standard library and the cubical library in the `release-2.6.3` branch. I don't know which should be the branch (and the commit) of the standard-library in `master`. I'll continue with the release later.

**Comment by MatthewDaggitt** [comment]
Apologies for the delay, I've updated the `v1.7.2` branch and pointed it at the head of `v2.6.3` branch.

> Great! However note that it won't fix the current issue with the benchmark-without-logs test.

Yes. Those dependencies on unreleased standard library code shouldn't have made it in to the `v2.6.3` branch so I think the changes introducing that dependency will have to backed out before release.

**Comment by asr** [comment]
Agda 2.6.3 has been released. Great job every one!

**Comment by nad** [comment]
> Is this not the `experimental` branch you're after?

Yes, more or less. The question is if the std-lib test should use this branch, or something like "v1.7.3".

**Comment by andreasabel** [comment]
I just realized 2.6.3 was released with `-Werror`.  
Not sure how this happened, I thought hackage would reject such a package.
But apparently, `cabal check` is happy with 2.6.3's `Agda.cabal`, so maybe it only complains if one has a global `-Werror`.
